cpp/src/arrow/util/span.h

```diff
@@ -25,6 +25,18 @@
 
 namespace arrow::util {
 
+template <class T>
+class span;
+
+template <class T, class R, class Enable = void>
+struct ConstructibleFromDataAndSize : std::false_type {};
+
+template <class T, class R>
```

Comment:
Can you add a comment explaining why this construct is needed, and pointing to the GH issue?

Reply:
Yup sure.

Reply:
Done.